src/index.ts

```diff
@@ -77,20 +78,70 @@ async function main () {
     manifest.addIndirectDependency(dep)
   })
 
-  const snapshot = new Snapshot(
-    {
+  type SnapshotDetector = {
+    name: string
+    url: string
+    version: string
+  }
+  let snapshotDetector: SnapshotDetector
+
+  const detectorName = core.getInput('detector-name')
+  const detectorUrl = core.getInput('detector-url')
+  const detectorVersion = core.getInput('detector-version')
+
+  if (detectorName === '' && detectorUrl === '' && detectorVersion === '') {
+    // use defaults if none are specified
+    snapshotDetector = {
       name: 'actions/go-dependency-submission',
       url: 'https://github.com/actions/go-dependency-submission',
       version: '0.0.1'
-    },
-    github.context,
```

Comment:
The `github.context` is where the sha and ref are taken from this in the snapshot constructor

https://github.com/github/dependency-submission-toolkit/blob/main/src/snapshot.ts#L126-L135

So that it does not need to be explicitly set